packages/marshal/src/marshal.js

```diff
@@ -484,6 +484,7 @@ export function makeMarshal(
     const rawTree = harden(JSON.parse(data.body));
     const fullRevive = makeFullRevive(data.slots);
     const result = harden(fullRevive(rawTree));
+    // See https://github.com/Agoric/agoric-sdk/issues/4302
```

Comment:
I'm actually still writing a more general issue for this.

Reply:
https://github.com/Agoric/agoric-sdk/issues/4337

Reply:
Done. Linked to https://github.com/Agoric/agoric-sdk/issues/4337 . But this PR only solves the specific #4302 problem.

Reply:
How so? Doesn't it prevent all things that produce non passable, like buggy convertSlotToVal ?

Reply:
***OOPS***. I just merged https://github.com/Agoric/agoric-sdk/pull/4304 which contained changes to two files in the `marshal` package of agoric-sdk. Thus, I made them out of sync.

@kriskowal How should we handle this?

Reply:
I noticed this in preparing to answer @mhofman 's question about #4337

In answer to that question, yes, the assert will catch that specific problem. But mysteriously. I'd like to put more checking directly around the convertSlotToVal call so I can complain with a more specific diagnostic. That same check can ensure that the result is only remotables and promises, which this PR does not enforce.

Reply:
Oh yes! Which is why I also wrote https://github.com/Agoric/agoric-sdk/issues/4336 for convertSlotToVal

https://github.com/Agoric/agoric-sdk/issues/4337 is the meta issue listing all known occurrences of non passables that this PR prevents.